node/v2/handler.js

```diff
@@ -90,6 +89,24 @@ TChannelV2Handler.prototype._write = function _write(frame, encoding, callback)
     }
 };
 
+TChannelV2Handler.prototype._read = function _read(/* n */) {
+    /* noop */
+};
+
+TChannelV2Handler.prototype.sendFrame = function(frame, callback) {
+    var self = this;
+    if (!callback) {
+        callback = function emitIt(err) {
+            if (err) self.emit('error', err);
+        };
+    }
+    if (self.push(frame)) {
+        callback();
+    } else {
+        self.once('drain', callback);
```

Comment:
Having read the source code, there is no good way to have a drain event.

The best you can do is listen to readable and check `self._readableState.length < self._readableState.highWaterMark`

There is no garantuee that the buffer is below the hwm after a readable in case the user calls `.read(2)` and only reads two bytes but we overfilled the buffer by 2MB.